packages/headless/src/features/commerce/context/context-state.ts

```diff
@@ -16,7 +17,7 @@ export const getContextInitialState = (): CommerceContextState => ({
   trackingId: '',
   language: '',
   country: '',
-  currency: '',
+  currency: '' as CurrencyCodeISO4217,
```

Comment:
If we want to avoid casting here, should we set a currency (like "USD")?

Reply:
Mh, I think it's OK with the cast

Reply:
Out of scope for this PR, but it's worth thinking about when these properties need to be set. They have to be in place before the first request. and they're all required. It feels like they are configuration options on the commerce engine rather than context?

If they were set on init, then we could set default values, avoid casting, and not worry about confusing a developer as to where `USD` came from. If we use context, then I'd avoid setting a default. Casting is ok, but avoiding the problem by making them config options feels like the better DX,

What do you think @Spuffynism?

Reply:
Great point!

Is the engine always created _after_ we know all the information in the context? If so, eliminating the context controller and replacing it by configuration options on the engine would make sense to me 🤔

I'm thinking in a SSR context, one might want to create the engine server-side, and then create the context controller client side, but I'm not sure that's an actual use case.

Reply:
One might want to update the values after the engine is created, e.g. a customer changes language mid session, so the controller would be useful in that scenario.

I think the values will be known before creating the engine. They have to be there before the engine can make he first request 🤷. This feels like the "primary" setup flow and would guarantee that the params are set before the first request, but the controller will still be helpful for implementations that are multi-language,